python/GafferSceneUI/SceneEditor.py

```diff
@@ -93,22 +96,39 @@ def _updateFromSet( self ) :
 					( p for p in GafferScene.ScenePlug.RecursiveOutputRange( node ) if not p.getName().startswith( "__" ) ),
 					None
 				)
+				if outputScenePlug is None:
+					outputImagePlug = next(
+						( p for p in GafferImage.ImagePlug.RecursiveOutputRange( node ) if not p.getName().startswith( "__" ) ),
+						None
+					)
+					if outputImagePlug:
+						try:
+							with self.context():
+								outputScenePlug = GafferScene.SceneAlgo.sourceScene( outputImagePlug )
+						except:
+							# The call to sourceScene could easily fail ( for example, because you're looking
+							# at an ImageReader with an invalid file path, so you can't read the metadata ).
+							# If the image is invalid, then there is no corresponding scene, and it's correct
+							# to leave outputScenePlug set to None
+							pass
+
 				if outputScenePlug is not None :
 					inputsToFill.pop( 0 ).setInput( outputScenePlug )
-					plugConnection = self.__parentingConnections.get( outputScenePlug )
+					plugConnection = self.__nodeConnections.get( outputScenePlug )
```

Comment:
Is it a bit misleading to index `__nodeConnections` with a plug perhaps? Would just `__connections` be any better?